### PR TITLE
Add a feature freeze test matrix

### PR DESCRIPTION
**Pull request by antalszava**
**Changes**

* Updates `compile.py` to have room for the release candidate workflow generation;
* Adds a `workflow-template-release-candidate.yml` example workflow file for the release candidate workflows;
* Generates the plugin specific workflow files using the updated `compile.py` and the `workflow-template-release-candidate.yml` example workflow file;
* Updates `README.md`.

**Benefits**

Adds `<plugin>-latest-rc` actions that run the latest version of the plugin with the release candidate branch of PennyLane.

See [latest built version](https://github.com/PennyLaneAI/plugin-test-matrix/blob/f99625cd16302fb97b3088e64b97f9cc3e2a1c44/README.md) and a screenshot (note Qiskit and SF pass with latest-latest, see the last point in this description):

Screenshot from 2022-02-01 19-40-55

**Potential drawbacks**

* The current approach hardcodes the exact release candidate branch (`v0.21.0-rc0`), so it needs to be updated for each release;
* There will be conflicts arising with [plugins that currently pin to `master`](https://github.com/PennyLaneAI/plugin-test-matrix/runs/5029977780?check_suite_focus=true#step:5:61) of PennyLane:
Screenshot from 2022-02-01 19-37-23

### CODE REVIEW

**Comment by antalszava** [comment]
Hi @josh146, tagging you for visibility on this (will ask for a review within our squad).

**Comment by josh146** [comment]
Good idea @antalszava!

File: README.md [hunk 1]
```diff
@@ -70,6 +70,25 @@ All entries in the matrix are tested against PennyLane latest (GitHub master).
   - **Latest/latest passing**: Nothing to be done! The plugin is ready to be released
     with the upcoming PennyLane release.
 
+## Feature freeze testing matrix
```
Comment (dime10) [inline README.md:73]: I guess there is no more space for an additional column in the matrix above?

**Review by dime10 — APPROVED** [review]
Looks good to me :)

**Comment by antalszava** [comment]
[sc-14403]

File: README.md (hunk 1 above)
Comment (antalszava) [reply README.md:73]: Good point! I might merge as is, it could be moved though.

**Pull request merged**